### DISCUSSION TITLE
Should i3 provide feedback when a regex fails to parse in a command criterion?

### DISCUSSION THREAD

**Discussion by attso** [i3/i3#5817] [Ideas]
I had encountered an interesting unexpected behavior recently. One of my firefox extensions opens up a pop up window in some cases. I wanted that window to float by default, unlike other firefox windows. I tried adding this rule:

```for_window [class="firefox" title="Extension: (Bitwarden"] floating enable```

Which at first seemed to be working. However, it was also floating all of my other firefox windows. After looking up the docs and realizing this was supposed to be a regex, I escaped the `(` and now it seems to be working as intended.

I found it a little unintuitive that when the criterion failed to parse, there was no error and the rest of the criteria was still valid. I'd suggest it would be more intuitive for there to be an error message, or for the entire criteria to be considered invalid.

**Comment by orestisfl** [comment]
I am not sure how this would work, regex criteria fail all the time because they are evaluated for every single window. You would be constantly getting error messages.
